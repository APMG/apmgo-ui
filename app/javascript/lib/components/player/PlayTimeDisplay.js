// @flow
import React from 'react'
import './AudioPlayer.scss'

export default function (props: {currentTime?: number}) {
  let { currentTime } = props

  if (!currentTime || isNaN(props.currentTime)) {
    return (<span styleName="currentTime">0:00</span>)
  }

  let minutes = Math.floor(currentTime / 60)
  let seconds = currentTime - (minutes * 60)
  let formatted = minutes + ':' + seconds.toString().padStart(2, '0')

  return (<span styleName="currentTime">{formatted}</span>)
}
