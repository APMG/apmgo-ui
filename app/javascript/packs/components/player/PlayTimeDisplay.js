// @flow
import React from 'react'
import { connect } from 'react-redux'

const PlayTimeDisplayPresenter = (props: {currentTime: number}) => {
  if (typeof props.currentTime === undefined || isNaN(props.currentTime)) {
    return (<div>0:00</div>)
  }

  let minutes = Math.floor(props.currentTime / 60),
      seconds = props.currentTime - (minutes * 60),
      formatted = minutes + ":" + seconds.toString().padStart(2, '0')

  return (<div>{formatted}</div>)
}
//
// const mapStateToProps = (state) => {
//   return {
//     currentTime: state.audioPlayer.currentTime
//   }
// }

export default PlayTimeDisplayPresenter //connect(mapStateToProps)(PlayTimeDisplayPresenter)
