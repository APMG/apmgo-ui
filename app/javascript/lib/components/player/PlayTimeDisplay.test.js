
import React from 'react'
import { getSnapshotJson } from '../../redux/__tests__/mock-initial-state'
import PlayTimeDisplay from './PlayTimeDisplay'

describe('Play Time Display Component', () => {
  describe('Presenter', () => {
    it('Works Without currentTime Prop and Displays 0:00', () => {
      let tree = getSnapshotJson(<PlayTimeDisplay />)
      expect(tree).toMatchSnapshot()
    })
    it('Displays currentTime When Prop is Passed', () => {
      let tree = getSnapshotJson(<PlayTimeDisplay currentTime={60} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
