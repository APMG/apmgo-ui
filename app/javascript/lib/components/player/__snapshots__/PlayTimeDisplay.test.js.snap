// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play Time Display Component Presenter Displays currentTime When Prop is Passed 1`] = `
<div>
  1:00
</div>
`;

exports[`Play Time Display Component Presenter Works Without currentTime Prop and Displays 0:00 1`] = `
<div>
  0:00
</div>
`;
